opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -208,8 +209,38 @@ def set_attribute(self, key: str, value: types.AttributeValue) -> None:
         if has_ended:
             logger.warning("Setting attribute on ended span.")
             return
+
+        if not isinstance(value, (int, float, bool, str, list, tuple)):
+            logger.warning("invalid type for attribute value")
+            return
+        if isinstance(value, (list, tuple)) and len(value) > 0:
+            return_code = self._check_sequence(value)
+            if return_code:
```

Comment:
This works but when comparing against `None`, `is` is [preferred](https://www.python.org/dev/peps/pep-0008/#programming-recommendations), so in this case it should read `if return_code is not None:`.